### PR TITLE
Add test for node-external-ip config parameter

### PR DESCRIPTION
**Pull request by manuelbuil**
Signed-off-by: Manuel Buil <mbuil@suse.com>

#### Proposed Changes ####

Add a test that verifies that two nodes can create a k8s cluster using their externalIPs

#### Types of Changes ####

New test

#### Verification ####

go test -v -timeout=15m tests/e2e/external/external_test.go

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/6358

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/e2e/external/external_test.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+var hardened = flag.Bool("hardened", false, "true or false")
+
+// Environment Variables Info:
+// E2E_RELEASE_VERSION=v1.23.1+k3s1 or nil for latest commit from master
+
+type objIP struct {
+	name string
+	ipv4 string
+	ipv6 string
+}
```
Comment (dereknola) [inline tests/e2e/external/external_test.go:33]: We should consolidate this under `testutils.go` and resue it across the `dualstack` and `external` test.

File: tests/e2e/external/external_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+func getPodIPs(kubeConfigFile string) ([]objIP, error) {
+	cmd := `kubectl get pods -A -o=jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.podIPs[*].ip}{"\n"}{end}' --kubeconfig=` + kubeConfigFile
+	return getObjIPs(cmd)
+}
+
+func getNodeIPs(kubeConfigFile string) ([]objIP, error) {
+	cmd := `kubectl get nodes -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.addresses[?(@.type == "InternalIP")].address}{"\n"}{end}' --kubeconfig=` + kubeConfigFile
+	return getObjIPs(cmd)
+}
```
Comment (dereknola) [inline tests/e2e/external/external_test.go:62]: Move these under `testutils.go` and use across both dualstack and external tests.

File: tests/e2e/external/external_test.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+		} else if len(fields) > 1 {
+			objIPs = append(objIPs, objIP{name: fields[0], ipv4: fields[1]})
+		} else {
+			objIPs = append(objIPs, objIP{name: fields[0]})
+		}
+	}
+	return objIPs, nil
+}
+
+func Test_E2EDualStack(t *testing.T) {
```
Comment (dereknola) [inline tests/e2e/external/external_test.go:86]:
```suggestion
func Test_E2EExternal(t *testing.T) {
```

**Review by dereknola — CHANGES_REQUESTED** [review]

File: tests/e2e/external/external_test.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
+// This test verifies that two nodes, which can't connect using the local network, are
+// able to still connect using the node-external-ip. In real life, node-external-ip
+// would be a public IP. In the test, we create two networks, one sets the node
+// internal-ip and the other sets the node-external-ip. Traffic is blocked on the former
+
+package validatenodeexternalip
```
Comment (dereknola) [inline tests/e2e/external/external_test.go:6]: You have the folder named `external` the test named `External-Ip Suite` the function named `E2EDualStack`.... can you consildate these all to the same name. 

Folder: `externalip`
Package: `externalip`
function: `E2EExternalIP`
suite: `External IP Suite`

**Review by dereknola — CHANGES_REQUESTED** [review]

File: tests/e2e/external/external_test.go (hunk 1 above)
Comment (manuelbuil) [reply tests/e2e/external/external_test.go:33]: good idea

File: tests/e2e/external/external_test.go (hunk 3 above)
Comment (manuelbuil) [reply tests/e2e/external/external_test.go:86]: oh! good catch!

File: tests/e2e/external/external_test.go (hunk 4 above)
Comment (manuelbuil) [reply tests/e2e/external/external_test.go:6]: good idea

**Review by dereknola — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**